2024/calcite-react/demo/src/Components/View/View.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useRef, useState } from "react"
+
+import WebMap from "@arcgis/core/WebMap";
+import MapView from "@arcgis/core/views/MapView";
+
+import "./View.scss";
+
+interface ViewProps {
+    webmap: string;
+}
+
+export const View = (props: ViewProps) => {
+    const viewRef = useRef(null) as React.RefObject<HTMLDivElement>;
+    const [ view, setView ] = useState<__esri.MapView | null>(null);
+
+    useEffect(() => {
+        return () => {
+            view?.destroy();
+        }
+    }, [])
+    
+    useEffect(() => {
+        if (!viewRef?.current) return;
+
+        const map = new WebMap({
+            portalItem: {
+                id: props.webmap
+            }
+        });
+
+        const view = new MapView({
+            container: viewRef?.current as HTMLDivElement,
+            map
+        });
+
+        setView(view);
+
```

Comment:
could you remove the useEffect above this one and inline the destruction function in here instead?

that would simplify the code a bit and improve clarity

```suggestion
    return () => view?.destroy();
```